streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -566,27 +566,51 @@ public int addRecords(final TopicPartition partition, final Iterable<ConsumerRec
     /**
      * Schedules a punctuation for the processor
      *
+     * @param startTime time of the first punctuation
      * @param interval the interval in milliseconds
      * @param type the punctuation type
      * @throws IllegalStateException if the current node is not null
      */
-    public Cancellable schedule(final long interval, final PunctuationType type, final Punctuator punctuator) {
+    public Cancellable schedule(final long startTime, final long interval, final PunctuationType type, final Punctuator punctuator) {
```

Comment:
As mentioned: this is a public API change and required a KIP. Please revert this for this PR (and feel free to propose a KIP for this change).